src/client/components/NotificationsMenu/NotificationsMenu.tsx

```diff
@@ -0,0 +1,86 @@
+import { useRouter } from "next/router";
+import React from "react";
+import { NotificationsIcon } from "./NotificationsIcon";
+import styles from "./NotificationsMenu.module.css";
+import cx from "classnames";
+import { useSession } from "next-auth/react";
+import { LoadingSpinner, useCustomTranslation } from "@coin-view/client";
+import { NotificationsMenuItem } from "@coin-view/client";
+
+
+
+export const NotificationsMenu = () => {
+  const { t } = useCustomTranslation();
+  const { data: session, status } = useSession();
+  const [menuOpen, setMenuOpen] = React.useState(false);
+  const [notifications, setNotifications] = React.useState<any[]>([]);
+  const [loading, setLoading] = React.useState(false);
+
+  const handleNotificationsClick = async () => {
+    setMenuOpen((open) => !open);
+    if(!menuOpen)
+    {
+      setLoading(true);
+      const response = await fetch("api/notifications");
+      const {error, notifications} = await response.json();
+      setLoading(false);
+      setNotifications(notifications);
+    }
+    else
+    {
+      setNotifications([]);
+    }
+  }
```

Comment:
No need to clear notifications, once they are loaded, they can stay in state. So add a condition if we have notifications, do not fetch api.
Wrap function in `useCallback` so it's not recreated every render